### PR TITLE
Use inline callbacks where possible (part 3)

### PR DESCRIPTION
**Pull request by mokibit**
Use of defer.inlineCallbacks leads to more readable code.

### CODE REVIEW

File: master/buildbot/test/unit/test_db_buildsets.py [hunk 1]
```diff
@@ -381,175 +351,163 @@ def insert_test_getRecentBuildsets_data(self):
...
-
-                # one buildset_sourcestamps row
-                r = conn.execute(self.db.model.buildset_sourcestamps.select())
-                self.assertEqual(
-                    list(r.keys()), [u'id', u'buildsetid', u'sourcestampid'])
-                self.assertEqual(r.fetchall(), [(1, bsid, 234)])
-            return self.db.pool.do(thd)
-        d.addCallback(check)
-        return d
+        xxx_todo_changeme = yield self.db.buildsets.addBuildset(
```
Comment (p12tic) [inline master/buildbot/test/unit/test_db_buildsets.py]: Just use (bsid, brids)

File: master/buildbot/test/unit/test_db_changes.py [hunk 2]
```diff
@@ -299,134 +283,100 @@ def insert7Changes(self):
...
+        # sort by changeid, since we assert on change 13 at index 0
+        changes.sort(key=lambda c: c['changeid'])
+        changeids = [c['changeid'] for c in changes]
+        self.assertEqual(changeids, [13, 14])
+        # double-check that they have .files, etc.
+        self.assertEqual(sorted(changes[0]['files']),
+                         sorted(['master/README.txt', 'worker/README.txt']))
+        self.assertEqual(changes[0]['properties'],
+                         {'notest': ('no', 'Change')})
+        yield self.db.changes.getChanges()
```
[the remark this answers is not in the document]
Comment (mokibit) [reply master/buildbot/test/unit/test_db_changes.py]: Fixed.

File: master/buildbot/test/unit/test_db_changes.py [hunk 3]
```diff
@@ -299,134 +283,100 @@ def insert7Changes(self):
...
-        d.addCallback(check)
-        d.addCallback(lambda _:
-                      self.db.changes.getChanges())
-        d.addCallback(check)
-        return d
+        changes = yield self.db.changes.getRecentChanges(5)
+
+        changeids = [c['changeid'] for c in changes]
+        self.assertEqual(changeids, [])
+        yield self.db.changes.getChanges()
```
[the remark this answers is not in the document]
Comment (mokibit) [reply master/buildbot/test/unit/test_db_changes.py]: Fixed.

File: master/buildbot/test/unit/test_db_buildsets.py (hunk 1 above)
Comment (mokibit) [reply master/buildbot/test/unit/test_db_buildsets.py]: Fixed.

**Pull request merged**